hal/stm32l4.c

```diff
@@ -0,0 +1,255 @@
+/* STM32L4 hal for wolfBoot 
+Using NVM_FLASH_WRITEONCE is required for the update to happen
+*/
+
+#include <stdint.h>
+#include <image.h>
+#include "stm32l4xx_hal.h"
+
+/* Assembly helpers */
+#define DMB() asm volatile ("dmb")
+
+/*** RCC ***/
+
+#define RCC_PRESCALER_DIV_NONE 0
+
+
+
+uint32_t Address = 0, PAGEError = 0;
+
+
+static FLASH_EraseInitTypeDef EraseInitStruct;
+  
+  
+static uint32_t GetPage(uint32_t Addr)
+{
+  uint32_t page = 0;
+  
+  if (Addr < (FLASH_BASE + FLASH_BANK_SIZE))
+  {
+    page = (Addr - FLASH_BASE) / FLASH_PAGE_SIZE;
+  }
+  else
+  {
+    page = (Addr - (FLASH_BASE + FLASH_BANK_SIZE)) / FLASH_PAGE_SIZE;
+  }
+  
+  return page;
+}
+
+
+static uint32_t GetBank(uint32_t Addr)
```

Comment:
make it RAMFUNCTION to avoid double long jump when called from `hal_flash_erase`